interlok-common/src/main/java/com/adaptris/interlok/resolver/FileResolver.java

```diff
@@ -0,0 +1,294 @@
+ *                  files as the resolve method returns a String).
+ *  - %size        - The file size (not meaningful for directories or
+ *                  other non-regular files).
+ *  - %type        - The type of file (regular file, directory, symlink,
+ *                  etc).
+ *  - %date_create - The date of file creation.
+ *  - %date_modify - The last time the files was modified.
+ *  - %date_access - The date the file was last accessed.
+ *  - %permissions - The file permissions (Unix-like; ugo-rwx).
+ * </p>
+ */
+public class FileResolver extends ResolverImp
+{
+  // Should match %file{file:%data%size%...}
+  private static final String RESOLVE_REGEXP = "^.*%file\\{(.+):(%[%a-z_]+)\\}.*$";
+  private transient Pattern resolverPattern;
+
+  private static final String DATE_FORMAT = "yyyy-MM-dd HH:mm:ss zzz";
+  private static final String SEPARATOR = ",";
+
+  @Getter
+  @Setter
+  @Valid
+  @InputFieldDefault(DATE_FORMAT)
+  @AdvancedConfig(rare = true)
+  private String dateFormat;
+
+  @Getter
+  @Setter
+  @Valid
+  @InputFieldDefault(SEPARATOR)
+  @AdvancedConfig(rare = true)
+  private String separator;
+
+  public FileResolver()
+  {
+    resolverPattern = Pattern.compile(RESOLVE_REGEXP, Pattern.DOTALL);
+  }
+
+  /**
+   * Attempt to resolve a value externally.
+   *
+   * @param lookupValue
+   * @return the resolved value
+   */
+  @Override
+  public String resolve(String lookupValue)
+  {
+    return resolve(lookupValue, (InterlokMessage)null);
+  }
+
+  @Override
+  public String resolve(String lookupValue, InterlokMessage target)
+  {
+    if (lookupValue == null)
+    {
+      return null;
+    }
+    String result = lookupValue;
+    log.trace("Resolving {} from filesystem", lookupValue);
+    Matcher m = resolverPattern.matcher(lookupValue);
+    while (m.matches())
+    {
+      File path;
+      String pathReplace = m.group(1);
+      // see if path is resolvable...
+      if (target != null)
+      {
+        path = new File(target.resolve(pathReplace));
+      }
+      else
+      {
+        path = new File(pathReplace);
+      }
+
+      String w = m.group(2);
+      StringBuffer sb = new StringBuffer();
+      for (String s : w.split("%"))
+      {
+        if (StringUtils.isEmpty(s))
+        {
+          continue;
+        }
+        What what = What.parse(s);
+        if (what == null)
+        {
+          log.error("{} is not something that's resolvable", s);
+          throw new UnresolvableException(s + " is not something that's resolvable");
+        }
+        log.trace("Resolve {} on path {} ", what, path);
+
+        String value = "";
+        try
+        {
+          value = what.resolve(path.toPath());
+          if (value == null)
+          {
+            log.warn("{} resolved to null for {}", what, path);
+            value = "";
+          }
+        }
+        catch (IOException e)
+        {
+          log.error("Could not resolve {} on path {} ", what, path, e);
+        }
```

Comment:
My only question would be, what's our expected behaviour if the file doesn't exist or the file is not "readable" by us.  If the answer is "undefined behaviour, possibly an exception" then that's probably cool, it's just a documentation issue...

Rather than logging an error and giving us something "meaningless" as the resolution (i.e. if the text was something like `%file{/path/exists:%size}-%file{/path/does/not/exist:%size}` we might end up with `%file{/path/does/not/exist:%size}` as the resolved text which could be completely unusable and untrackable.

You're "better off" failing with a UnresolveableException in all cases?